### DISCUSSION TITLE
How do I use Apexcharts to display how long Nest heater was heating/cooling during time period

### DISCUSSION THREAD

**Discussion by GCV-Sleeper-Service** [RomRider/apexcharts-card#700] [Q&A]
I created a helper in HA to see when Nest thermostat was on. When I add this template sensor to history dashboard, it shows specific action and duration:
![image](https://github.com/RomRider/apexcharts-card/assets/45667495/46963ba9-5c5b-4a1c-928c-0d83ce05cf97)

How can I use Apexchart to summarize how long the heating was on during today, previous day, last week, last month and since Jan 1st?

**Comment by caiusseverus** [comment]
You can make a template sensor to check whether the nest is heating or not:

```
  - sensor:
      - name: heating_activity
        state: "{{ state_attr('climate.nest', 'hvac_action') }}"
```
You can use the history stats platform to record time in a particular state. For example to get the hours heating today:
```
  - platform: history_stats
    name: Heating today
    entity_id: sensor.heating_activity
    state: "heating"
    type: time
    start: "{{ now().replace(hour=0, minute=0, second=0) }}"
    end: "{{ now() }}"
```

Then you create a utility meter helper for the above sensor with the appropriate reset cycle you want, and you can use that as a sensor for graphing etc.
